### PR TITLE
Raise exception for invalid kwargs in Metric base class

### LINKED ISSUE #1426: Metric base class does not check for extra kwargs
## 🐛 Bug

The `Metric` base class does not check for extra kwargs [here](https://github.com/Lightning-AI/metrics/blob/bc057d495703aff5c4d8d671c54060e08a677142/src/torchmetrics/metric.py#L94-L120).
Users can currently do something like `torchmetrics.MeanMetric(foo=True)` without issue.

### To Reproduce

```
import torchmetrics
torchmetrics.MeanMetric(your_favorite_word="lightning")
```

### Expected behavior

Raise an exception if invalid kwargs are passed

### Environment

Latest TorchMetrics version. No other dependencies matter.

### Additional context

[Slack thread](https://pytorch-lightning.slack.com/archives/C010R4A7189/p1672859410056429)

### PR DESCRIPTION
**Pull request by EPronovost**
## What does this PR do?

Fixes #1426 

## Before submitting

- [X] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [X] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [X] Did you write any new **necessary tests**?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/torchmetrics/metric.py [hunk 1]
```diff
@@ -119,6 +119,9 @@ def __init__(
                 f"Expected keyword argument `sync_on_compute` to be a `bool` but got {self.sync_on_compute}"
             )
 
+        if kwargs:
+            raise ValueError("Unexpected keyword arguments: " + " ".join(sorted(kwargs)))
```
Comment (Borda) [inline src/torchmetrics/metric.py:123]:
```suggestion
            kwargs_ = [f'`{a}`' for a in sorted(kwargs)]
            raise ValueError(f"Unexpected keyword arguments: {', '.join(kwargs_)}")
```

File: tests/unittests/bases/test_metric.py [hunk 2]
```diff
@@ -44,6 +44,12 @@ def test_error_on_wrong_input():
     with pytest.raises(ValueError, match="Expected keyword argument `compute_on_cpu` to be an `bool` bu.*"):
         DummyMetric(compute_on_cpu=None)
 
+    with pytest.raises(ValueError, match="Unexpected keyword arguments: foo"):
+        DummyMetric(foo=True)
+
+    with pytest.raises(ValueError, match="Unexpected keyword arguments: bar foo"):
```
Comment (Borda) [inline tests/unittests/bases/test_metric.py:50]:
```suggestion
    with pytest.raises(ValueError, match="Unexpected keyword arguments: `bar`, `foo`"):
```

**Review by Borda — APPROVED** [review]

File: src/torchmetrics/metric.py [hunk 3]
```diff
@@ -119,6 +119,10 @@ def __init__(
                 f"Expected keyword argument `sync_on_compute` to be a `bool` but got {self.sync_on_compute}"
             )
 
+        if kwargs:
```
Comment (Borda) [inline src/torchmetrics/metric.py:122]: ok, this need some adjustment as you can see failing integrations tests :)

**Review by Borda — CHANGES_REQUESTED** [review]

**Review by SkafteNicki — APPROVED** [review]
Fixed the integration.
The integration test had the exact problem mentioned in the issue.

**Review by Borda — APPROVED** [review]
